### PR TITLE
Get basic image metadata from image properties

### PR DESCRIPTION
**Pull request by Inokinoki**
Generate meta data from images.

### CODE REVIEW

**Comment by Inokinoki** [comment]
Should we add them automatically in any datasets?

File: giskard_vision/core/detectors/base.py [hunk 1]
```diff
@@ -169,3 +169,13 @@ def get_results(self, model: Any, dataset: Any) -> List[ScanResult]:
...
+
+
+EthicalIssueMeta = IssueGroup(
+    "Ethical",
+    description="The data are filtered by metadata like age, facial hair, or gender to detect ethical biases.",
+)
+PerformanceIssueMeta = IssueGroup(
+    "Performance",
+    description="The data are filtered by metadata like emotion, head pose, or exposure value to detect performance issues.",
+)
```
Comment (rabah-khalek) [inline giskard_vision/core/detectors/base.py:181]: these are now in `dataloaders/base.py`, they also belong nicely here, but as the issue groups are relevant for dataloaders when setting meta, i would leave it the dataloaders base, could you refactor this please?

File: giskard_vision/core/dataloaders/base.py [hunk 2]
```diff
@@ -119,10 +128,48 @@ def meta_none(self) -> Optional[TypesBase.meta]:
         Returns default for meta data if it is None.
 
         Returns:
-            Optional[np.ndarray]: Default for meta data.
+            Optional[TypesBase.meta]: Default for meta data.
         """
         return None
 
+    def meta_auto(self, idx: int) -> TypesBase.meta:
```
Comment (rabah-khalek) [inline giskard_vision/core/dataloaders/base.py:135]: let's move this to `get_meta` as we want them to available for all dataloaders by default. 
That means we should change all daughter class `get_meta` in order to append the dataset-specific meta-data to these default ones.

File: giskard_vision/core/dataloaders/hf.py [hunk 3]
```diff
@@ -101,3 +111,37 @@ def cleanup(self):
         Clean the temporary folder
         """
         shutil.rmtree(self.temp_folder)
+
+    @abstractmethod
+    def get_raw_hf_image(self, idx: int) -> PILImage:
```
Comment (rabah-khalek) [inline giskard_vision/core/dataloaders/hf.py:116]: are images always in the PIL type from HF?

**Review by rabah-khalek — CHANGES_REQUESTED** [review]
overall good, let's just promot meta_auto to the default get_meta method

File: giskard_vision/core/detectors/base.py (hunk 1 above)
Comment (Inokinoki) [reply giskard_vision/core/detectors/base.py:181]: ofc

File: giskard_vision/core/dataloaders/base.py (hunk 2 above)
Comment (Inokinoki) [reply giskard_vision/core/dataloaders/base.py]: Done

File: giskard_vision/core/dataloaders/hf.py (hunk 3 above)
Comment (Inokinoki) [reply giskard_vision/core/dataloaders/hf.py:115]: Yes, it decodes any formats and return a PIL image to use in HF pipeline

File: giskard_vision/core/detectors/base.py (hunk 1 above)
Comment (Inokinoki) [reply giskard_vision/core/detectors/base.py:181]: The user can call the method in the super class to get the basic metadata, and then filter them or change the issue group during the merge of the dataset-specific ones.

**Review by rabah-khalek — APPROVED** [review]
lgtm, let's just change the issue group of the default meta

**Pull request merged**